src/Client/Native/NativeNip17.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Revolution\Nostr\Client\Native;
+
+use Exception;
+use Illuminate\Container\Container;
+use Illuminate\Contracts\Container\BindingResolutionException;
+use Illuminate\Http\Client\Response;
+use Illuminate\Support\Traits\Conditionable;
+use Illuminate\Support\Traits\Macroable;
+use Revolution\Nostr\Client\Native\Concerns\HasHttp;
+use Revolution\Nostr\Contracts\Client\ClientNip17;
+use swentel\nostr\Key\Key;
+use swentel\nostr\Nip17\DirectMessage;
+use swentel\nostr\Nip59\GiftWrapService;
+use swentel\nostr\Sign\Sign;
+
+class NativeNip17 implements ClientNip17
+{
+    use Conditionable;
+    use HasHttp;
+    use Macroable;
+
+    /**
+     * Send a private direct message using NIP-17.
+     *
+     * @param  string  $sk  sender's secret key
+     * @param  string  $pk  receiver's public key
+     * @param  string  $message  message content
+     * @param  array  $additionalTags  additional tags to include
+     * @param  string|null  $replyToId  ID of message being replied to (optional)
+     *
+     * @throws BindingResolutionException
+     * @throws Exception
+     */
+    public function sendDirectMessage(
+        #[\SensitiveParameter] string $sk,
+        string $pk,
+        string $message,
+        array $additionalTags = [],
+        ?string $replyToId = null,
+    ): Response {
+        $key = Container::getInstance()->make(Key::class);
+        $sign = Container::getInstance()->make(Sign::class);
+        $giftWrapService = Container::getInstance()->makeWith(GiftWrapService::class, [$key, $sign]);
+        $directMessage = Container::getInstance()->makeWith(DirectMessage::class, [$giftWrapService, $key]);
```

Comment:
The usage of makeWith is `->makeWith(GiftWrapService::class, ['key' => $key, 'sign' => $sign])`